src/PKSim.Core/Snapshots/Mappers/PopulationMapper.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using OSPSuite.Utility.Exceptions;
+using PKSim.Assets;
+using PKSim.Core.Model;
+using PKSim.Core.Repositories;
+using SnapshotPopulation = PKSim.Core.Snapshots.Population;
+using ModelPopulation = PKSim.Core.Model.Population;
+
+namespace PKSim.Core.Snapshots.Mappers
+{
+   public class PopulationMapper : ObjectBaseSnapshotMapperBase<ModelPopulation, SnapshotPopulation>
+   {
+      private readonly IndividualMapper _individualMapper;
+      private readonly ParameterRangeMapper _parameterRangeMapper;
+      private readonly AdvancedParameterMapper _advancedParameterMapper;
+      private readonly IGenderRepository _genderRepository;
+
+      public PopulationMapper(
+         IndividualMapper individualMapper,
+         ParameterRangeMapper parameterRangeMapper,
+         AdvancedParameterMapper advancedParameterMapper,
+         IGenderRepository genderRepository)
+      {
+         _individualMapper = individualMapper;
+         _parameterRangeMapper = parameterRangeMapper;
+         _advancedParameterMapper = advancedParameterMapper;
+         _genderRepository = genderRepository;
+      }
+
+      public override SnapshotPopulation MapToSnapshot(ModelPopulation population)
+      {
+         return SnapshotFrom(population, snapshot =>
+         {
+            snapshot.Seed = population.Seed;
+            mapPopulationProperties(snapshot, population);
+         });
+      }
+
+      private void mapPopulationProperties(SnapshotPopulation snapshot, ModelPopulation population)
+      {
+         switch (population)
+         {
+            case RandomPopulation randomPopulation:
+               mapRandomPopulationProperties(snapshot, randomPopulation);
+               break;
+            default:
+               throw new OSPSuiteException(PKSimConstants.Error.PopulationSnapshotOnlySupportedForRandomPopulation);
```

Comment:
Exception if we are not dealing with a random population. This is an OSPSuiteException which will result in NO stack trace

Reply:
@Yuri05 Do you agree?